sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py

```diff
@@ -62,6 +62,10 @@ def send(self, events, **kwargs):
             events = [events]
 
         if all(isinstance(e, CloudEvent) for e in events) or all(_is_cloud_event(e) for e in events):
+            try:
+                events = [e._to_generated(**kwargs) for e in events]
+            except AttributeError:
+                pass # means it's a dictionary
```

Comment:
how?

Reply:
@rakshith91 still, how could this happen?

Reply:
sorry  - missed it 
dictionary will not have a _to_generated - so we do nothing and assume it's in the right format